docs/source/srs.rst

```diff
@@ -0,0 +1,283 @@
+processing of data and manages CAN network communication for the IMU. The
+BNO055 is used to obtain the orientation measurements.
+
+* list some boards that will use the data
+* difference between linear acceleration data and accelerometer data?
+
+User Interfaces
+~~~~~~~~~~~~~~~
+
+Users will rarely interact directly with the DEV1 IMU software. The DEV1 IMU 
+software will mainly be interfaced with via CANopen and thus will require 
+additional tools to interact with the DEV1 IMU. There is no current plan 
+for a team-developed tool to provide an interface for interaction with the 
+DEV1 IMU.
+
+Hardware Interfaces
+~~~~~~~~~~~~~~~~~~~
+
+The DEV1 IMU is connected to the bike's CANopen network which is made up of a
+two-wire differential pair. The on-board ST microcontroller is programmable
+through JTAG.
+
+Software Interfaces
+~~~~~~~~~~~~~~~~~~~
+
+The DEV1 IMU software interface is the report of measurements over the CAN
+network. Note that configurations and functionality of the IMU cannot be
+adjusted or controlled by other boards over CAN.
+
+Communication Interfaces
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+There is no communication between IMU and the other boards on the DEV1 bike ...
+
+* using a strict definition of communication as in both parties are connected
+at the same time
+
+Communication between the STM32F334r8 microcontroller and the BNO055 are
+handled via I2C, however it does not adhere to standard I2C. For more about
+the non-standard I2C communication scheme, read ...
+
+* should leave implementation of i2c in the state of software doc?
+
+Memory Constraints
+~~~~~~~~~~~~~~~~~~
+
+The produced software is limited to the 64KB of flash memory that is available
+on the STM32F334r8. Therefore the resulting binary must fit within this size.
+
+Operations
+~~~~~~~~~~
+
+The DEV1 IMU's main operation is to relay data measured by the BNO055
+sensor to the rest of the bike through the CANopen network. This includes
+initialization and configuration of the BNO055 chip, and managing the
+CANopen network communication.
+
+Site Adaptation
+~~~~~~~~~~~~~~~
+
+The DEV1 IMU is intended specifically for the DEV1 system. Therefore, the 
+software requirements and design will center around the specifics of the DEV1 
+system. No additional adaptations are currently being considered.
+
+Product Functions
+-----------------
+
+Initialization and Configuration of BNO055
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+One of the main component of the DEV1 IMU is the BNO055 chip, which requires
+proper initialization and configuration before it will collect data. The
+following procedure is informed by the BNO055's datasheet which details
+correct usage of the chip. After establishing I2C connection with the chip,
+the driver class will:
+
+* check that the BNO055's self-test initiated at startup was successful
+* switch the BNO055 from configuration mode to NDOF operation mode
+
+The NDOF operation mode calculates fused absolute orientation data from the
```

Comment:
Not sure why should it be 9DOF - is it a standard name? The code and documentation both have it as NDOF so I thought it would be best to have it with the same name

Reply:
If that matches the documentation, match the documentation. I normally see 6 degree-of-freedom stuff marked as 6DOF, but it's better to follow the standard set by the manufacturer